analyses/focal-cn-file-preparation/01-add-ploidy-cnvkit.Rmd

```diff
@@ -26,10 +26,63 @@ histologies_file <- file.path("..", "..", "data", "histologies.tsv")
 histologies_df <- readr::read_tsv(histologies_file, guess_max = 100000)
 ```
 
+# Edits for https://github.com/PediatricOpenTargets/ticket-tracker/issues/177
+```{r}
+# Filter to WXS tumor with no germline sex estimate
+no_estimate <- histologies_df %>% 
+  filter(sample_type == "Tumor") %>% 
+  filter(experimental_strategy == "WXS") %>%  
+  filter(is.na(germline_sex_estimate)) %>%
+  select(-germline_sex_estimate)
+
+# Extract participant ID with no germline sex estimate
+no_estimate_participants <- no_estimate %>% 
+  pull(cohort_participant_id) %>% 
+  unique()
+
+# Find out matching WGS - currently this has no entry for v9
+no_estimate_participants_match_wgs <- histologies_df %>% 
+  filter(cohort_participant_id %in% 
+  no_estimate_participants) %>% 
+  filter(experimental_strategy == "WGS") %>% 
+  filter(sample_type == "Tumor") %>% 
+  select(cohort_participant_id, germline_sex_estimate)
+
+# Merge data that was fixed with WGS germline sex estimate
+no_estimate_fixed_through_wgs <- left_join(no_estimate, no_estimate_participants_match_wgs, by = "cohort_participant_id") %>% 
+  filter(!is.na(germline_sex_estimate))
+
+# Identify data that was not fixed due to NA
+no_estimate_still_na <- left_join(no_estimate, no_estimate_participants_match_wgs
+                                  , by = "cohort_participant_id") %>%
+  filter(is.na(germline_sex_estimate))
+
+# Fix NA data using gender
+no_estimate_still_na_fixed <- no_estimate_still_na %>% 
+  mutate(germline_sex_estimate = case_when(
+                                  reported_gender == "Female" ~ "Female", 
+                                  reported_gender == "Male" ~ "Male",
+                                  TRUE ~ as.character(NA))
+         )
+
+# Merge back fixed_wgs, fixed_na, and exist_germline_estimate data
+histologies_df_fixed <- bind_rows(no_estimate_fixed_through_wgs, no_estimate_still_na_fixed)
+
+histologies_df_final <- histologies_df %>% 
```

Comment:
Here, I am getting the same n as fixed, so this merge doesn't seem to be working properly

```
> table(histologies_df_final$germline_sex_estimate)

Female   Male 
   155    216 
> table(histologies_df$germline_sex_estimate)

 Female   known    Male Unknown 
   1629      16    1892       1
```
I think this should be Female 1629+155 and Male 1892+216 - cc @runjin326

Reply:
I think what you want to do here after:
```
# Merge back fixed_wgs, fixed_na, and exist_germline_estimate data
histologies_df_fixed <- bind_rows(no_estimate_fixed_through_wgs, no_estimate_still_na_fixed)
```

 is this:

```
# remove samples in histologies_df_fixed from histologies_df
histologies_df_rm <- histologies_df %>%
  anti_join(histologies_df_fixed, by = "Kids_First_Biospecimen_ID")

# add new rows back
histologies_df_final <- histologies_df_rm %>% 
  bind_rows(histologies_df_fixed)
```

Now I get the expected N of germline sex estimate:

```
> table(histologies_df_final$germline_sex_estimate)

 Female   known    Male Unknown 
   1784      16    2108       1 
> 1629+155
[1] 1784
> 1892+216
[1] 2108
```

Reply:
@sangeetashukla and @runjin326 added a comment about this [here](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/95/files#r702130039)

Reply:
@jharenza @runjin326 
It makes sense to use KF participant ID as you pointed out.
I am now using cohort_participant_id only to identify samples that have WGS performed when WXS is also performed.

However, while the split between the histologies_df and histologies_df_rm for KF Biospecimen ID looks correct; the split between male/female values for GSE in histologies_df and histologies_df_rm is the same. That is confusing to me. Any thoughts?

Reply:
>I am now using cohort_participant_id only to identify samples that have WGS performed when WXS is also performed.

We should not use `cohort_participant_id` at all - can you update this please?